Use the same simulated iPhone model for layout tests an API tests

ryanhaddad opened the pull request:
#### 123c6f6b61555beda2a7c90cbec50ca77e8b8ede
<pre>
Use the same simulated iPhone model for layout tests an API tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=239885">https://bugs.webkit.org/show_bug.cgi?id=239885</a>
&lt;rdar://92505373&gt;

Reviewed by NOBODY (OOPS!).

Something is different about the way layout test and API test managers
request iOS simulators, which causes API tests to use the first available
model as opposed to the default device types specified for the iOS
Simulator port.

* Tools/Scripts/webkitpy/api_tests/manager.py:
(Manager._initialize_devices): As what might be a temporary workaround until
someone with more webkitpy knowledge can take a look, explicitly request
the device types that match our defaults for iPhone and iPad simulators.
</pre>

JonWBedard commented on Tools/Scripts/webkitpy/api_tests/manager.py:
```diff
@@ -150,8 +151,10 @@ def _print_tests_result_with_status(self, status, runner):
-            SimulatedDeviceManager.initialize_devices(DeviceRequest(self._port.DEVICE_TYPE, allow_incomplete_match=True), self.host, simulator_ui=False)
+        if any(port in self._port.port_name for port in ['iphone-simulator', 'ios-simulator']):
+            SimulatedDeviceManager.initialize_devices(DeviceRequest(DeviceType(hardware_family='iPhone', hardware_type='12'), allow_incomplete_match=False), self.host, simulator_ui=True)
+        if 'ipad-simulator' in self._port.port_name:
```
Should be `elif` here, right?

JonWBedard commented on Tools/Scripts/webkitpy/api_tests/manager.py:
```diff
@@ -150,8 +151,10 @@ def _print_tests_result_with_status(self, status, runner):
 
     def _initialize_devices(self):
-        if 'simulator' in self._port.port_name:
-            SimulatedDeviceManager.initialize_devices(DeviceRequest(self._port.DEVICE_TYPE, allow_incomplete_match=True), self.host, simulator_ui=False)
```
Maybe we should keep this around as an `elif`, otherwise we're breaking watchOS. I know at least some API tests worked on watchOS at some point, although infrastructure doesn't run it.

ryanhaddad committed:
Use the same simulated iPhone model for layout tests an API tests
https://bugs.webkit.org/show_bug.cgi?id=239885
<rdar://92505373>

Reviewed by NOBODY (OOPS!).

Something is different about the way layout test and API test managers
request iOS simulators, which causes API tests to use the first available
model as opposed to the default device types specified for the iOS
Simulator port.

* Tools/Scripts/webkitpy/api_tests/manager.py:
(Manager._initialize_devices): As what might be a temporary workaround until
someone with more webkitpy knowledge can take a look, explicitly request
the device types that match our defaults for iPhone and iPad simulators.
